lms/static/scripts/frontend_apps/components/dashboard/test/SyncGradesButton-test.js

```diff
@@ -0,0 +1,181 @@
+import {
+  checkAccessibility,
+  mockImportedComponents,
+} from '@hypothesis/frontend-testing';
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+import sinon from 'sinon';
+
+import { Config } from '../../../config';
+import SyncGradesButton, { $imports } from '../SyncGradesButton';
+
+describe('SyncGradesButton', () => {
+  let fakeConfig;
+  let fakeApiCall;
+  let fakeUsePolledAPIFetch;
+
+  const studentsToSync = [
+    { h_userid: '123', grade: 50 },
+    { h_userid: '456', grade: 20 },
+  ];
+
+  beforeEach(() => {
+    fakeApiCall = sinon.stub();
+    fakeUsePolledAPIFetch = sinon.stub().callsFake(({ shouldRefresh }) => {
+      // Calling `shouldRefresh` just to hit required code coverage
+      shouldRefresh({
+        data: { status: 'scheduled' },
+      });
```

Comment:
It would be preferable to check that the logic in the `shouldRefresh` callback is actually correct. This seems like it should be quite feasible.